### PR TITLE
Fix math point types

### PR DESCRIPTION
**Pull request by Rapen765**
Changed functions' return type from `tuple[float, float]` to `Point2`.
Fixed documentation of `rand_vec_magnitude` and `rand_on_circle`.

### CODE REVIEW

File: arcade/math.py [hunk 1]
```diff
@@ -304,12 +301,12 @@ def rand_vec_magnitude(
     angle: float,
     lo_magnitude: float,
     hi_magnitude: float,
-) -> tuple[float, float]:
+) -> Point2:
     """
-    Returns a random vector, within a spread of the given angle.
+    Returns a random vector, with random magnitude.
```
Comment (pushfoo) [inline arcade/math.py]:
```suggestion
    Return a vector of randomized magnitude pointing in the given direction.
```

File: arcade/math.py [hunk 2]
```diff
@@ -304,12 +301,12 @@ def rand_vec_magnitude(
...
 
     Args:
-        angle (float): The angle to spread from
+        angle (float): The vector angle
```
Comment (pushfoo) [inline arcade/math.py]:
```suggestion
        angle (float): The vector angle in radians
```

**Review by pushfoo — CHANGES_REQUESTED** [review]
**TL;DR:** I'm undecided on the `Point2` change, but I have suggestions on the doc and the doc

@DragonMoffon any thoughts on consistency for `Point2` vs the perf dangers of `Vec2`?

@Rapen765 The CI's linter complaints. Could you please try the following in your local repo clone?
1. Make sure your venv's activated
2. `./make.py format && ./make.py ruff`
3. Commit & push

**Comment by DragonMoffon** [comment]
I def think Point2 should be used over tuple[float, float] everywhere. The only place it shouldn't be used is anything that returns a `tuple[int, int]` that should have a different alias (`Size2` maybe)

**Review by pushfoo — APPROVED** [review]

**Comment by pushfoo** [comment]
Merging to nudge velocity back up for 3.0.

**Comment by pushfoo** [comment]
Ty for the contribution and sorry for the wait!
